scripts/generate-types.js

```diff
@@ -14,10 +14,19 @@ const ORIGINAL_TYPES_PATH = "src_types";
 const DEST_TYPES_PATH = "types";
 const TYPES_COMMON_PATH = path.join(DEST_TYPES_PATH, "common");
 const TYPES_PAGES_PATH = path.join(DEST_TYPES_PATH, "pages");
+const DOCS_BRANCH = process.env.DOCS_BRANCH;
+
+const getGitCloneOptions = tmpDir => {
+  const gitOptions = ["clone"];
+  if (DOCS_BRANCH) {
+    gitOptions.push(...["--single-branch", "--branch", DOCS_BRANCH]);
+  }
+  return gitOptions.concat([WIX_CODE_DOCS_REMOTE, tmpDir.name]);
+};
```

Comment:
it feels cumbersome separating "git clone options" from the rest.
i think i would leave all the logic in the "cloneDocsRepo" function and try to only put the branch part under a condition.